builtin/aws/ecs/task.go

```diff
@@ -209,24 +212,152 @@ func (p *TaskLauncher) Config() (interface{}, error) {
 	return &p.config, nil
 }
 
-// StopTask signals to docker to stop the container created previously. This
-// method is currently unimplemented.
+// StopTask signals to AWS ECS to stop the container created previously.
 func (p *TaskLauncher) StopTask(
 	ctx context.Context,
 	log hclog.Logger,
 	ti *TaskInfo,
 ) error {
-	log.Warn("the StopTask method is not currently implemented for ECS tasks")
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Region: p.config.Region,
+	})
+	if err != nil {
+		return err
+	}
+
+	ecsSvc := ecs.New(sess)
+	tasks, err := ecsSvc.ListTasks(&ecs.ListTasksInput{
+		Cluster: aws.String(p.config.Cluster),
+		Family:  aws.String("waypoint-runner"),
+	})
+	if err != nil {
+		return err
+	}
+	for _, taskArn := range tasks.TaskArns {
+		taskResp, err := ecsSvc.DescribeTasks(&ecs.DescribeTasksInput{
+			Cluster: aws.String(p.config.Cluster),
+			Tasks:   aws.StringSlice([]string{*taskArn}),
+			Include: aws.StringSlice([]string{"TAGS"}),
+		})
+		if err != nil {
+			return err
+		} else if len(taskResp.Tasks) != 1 {
+			return status.Errorf(codes.Internal, "there should be only 1 task, but there are %d", len(taskResp.Tasks))
+		}
+		for _, tag := range taskResp.Tasks[0].Tags {
+			if *tag.Key == "waypoint-odr-task-name" && *tag.Value == ti.Id {
+				log.Info("stopping ECS task", "task_id", ti.Id)
+				_, err := ecsSvc.StopTask(&ecs.StopTaskInput{
+					Cluster: aws.String(p.config.Cluster),
+					Reason:  aws.String("Waypoint ODR job is complete"),
+					Task:    taskArn,
+				})
+				if err != nil {
+					return err
+				}
+				return nil
+			}
+		}
+	}
+
+	// If we reach this point, then we could not find our task in ECS
+	// so there's nothing to clean up. Normally, ECS will automatically
+	// clean up the task.
 	return nil
 }
 
 // WatchTask implements TaskLauncher
 func (p *TaskLauncher) WatchTask(
 	ctx context.Context,
 	log hclog.Logger,
+	ui terminal.UI,
 	ti *TaskInfo,
 ) (*component.TaskResult, error) {
-	return nil, status.Errorf(codes.Unimplemented, "WatchTask not implemented")
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Region: p.config.Region,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// this channel will receive the status of the ECS task
+	taskStatusCh := make(chan string)
+	go taskStatus(ctx, time.Duration(5*time.Minute), &taskStatusCh, log, sess, p.config.Cluster, ti.Id)
```

Comment:
Maybe it makes sense to have `taskStatus` with a pointer recevier (i.e. `func (p *TaskLauncher) taskStatus (...)`). Then we could reduce some of these parameters?